src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -602,10 +533,23 @@
             (catch RaiseException e
               (handle-task-details-jruby-exception e environment module task))))))
 
+(defn info-service
+  [request]
+  (let [path-components (-> request :route-info :path)
+        ;; path-components will be something like
+        ;; ["/puppet" "/v3" "/environment_classes" ["*" :rest]]
+        ;; and we want to map "/environment_classes" to a
+        ;; cacheable info service
+        resource-component (-> path-components butlast last)]
+    (when resource-component
+      (get {"environment_classes" :classes
+            "environment_transports" :transports}
+           (str/replace resource-component "/" "")))))
```

Comment:
you could have just put the / in the strings in the map

Reply:
the slash seemed incidental to how we were doing the routing rather than the semantics of the route segment. it be weird for this to break if the segments changed from "/environment_classes" to "environment_classes" or "environment_classes/". but maybe that's yagni?

Reply:
you can do it like this too it's not a big deal, just seemed like it added an extra call in here that you didn't need

Reply:
i don't care that much either way